### PR TITLE
UF-459: Extract Preference API from extensions

### PR DESCRIPTION
**Pull request by adrielparedes**
@paulovmr I've extracted the API, please, check everything is correct and nothing is missing. There are many other projects associated to this change

### CODE REVIEW

**Comment by adrielparedes** [comment]
jenkins retest this

**Comment by ederign** [comment]
jenkins retest this

File: uberfire-extensions/uberfire-wires/uberfire-wires-webapp/src/main/java/org/uberfire/ext/wires/client/preferences/example/MyInnerPreferenceCustomForm.java [hunk 1]
```diff
@@ -23,12 +23,10 @@
 import org.uberfire.client.annotations.WorkbenchPartView;
 import org.uberfire.client.annotations.WorkbenchScreen;
 import org.uberfire.client.mvp.UberElement;
-import org.uberfire.ext.preferences.client.annotations.PreferenceForm;
 import org.uberfire.ext.preferences.client.base.BasePreferenceForm;
 import org.uberfire.ext.wires.shared.preferences.bean.MyInnerPreference;
 
 @Dependent
-@PreferenceForm("MyInnerPreference")
```
Comment (paulovmr) [inline uberfire-extensions/uberfire-wires/uberfire-wires-webapp/src/main/java/org/uberfire/ext/wires/client/preferences/example/MyInnerPreferenceCustomForm.java]: Why was this removed?

File: uberfire-preferences/uberfire-preferences-api/src/main/java/org/uberfire/preferences/shared/PreferenceStore.java [hunk 2]
```diff
@@ -238,8 +238,8 @@
      * @return A preference scoped value, which contains the preference value and its scope (or null,
      * if the preference does not exist).
      */
-    <T> PreferenceScopedValue<T> getScoped( PreferenceScopeResolutionStrategyInfo scopeResolutionStrategyInfo,
-                                            String key );
+    <T> PreferenceScopedValue<T> getScoped(PreferenceScopeResolutionStrategyInfo scopeResolutionStrategyInfo,
```
Comment (paulovmr) [inline uberfire-preferences/uberfire-preferences-api/src/main/java/org/uberfire/preferences/shared/PreferenceStore.java]: Minor format issue (space missing after "(").

File: uberfire-preferences/uberfire-preferences-api/src/main/java/org/uberfire/preferences/shared/bean/mock/PortablePreferenceMockBeanGeneratedImpl.java [hunk 3]
```diff
@@ -84,7 +84,7 @@ public void load( final PreferenceScopeResolutionStrategyInfo customScopeResolut
                     errorCallback );
     }
 
-    private ParameterizedCommand<BasePreferencePortable<PortablePreferenceMock>> getLoadSuccessCallback( final ParameterizedCommand<PortablePreferenceMock> successCallback ) {
+    private ParameterizedCommand<BasePreferencePortable<PortablePreferenceMock>> getLoadSuccessCallback(final ParameterizedCommand<PortablePreferenceMock> successCallback ) {
```
Comment (paulovmr) [inline uberfire-preferences/uberfire-preferences-api/src/main/java/org/uberfire/preferences/shared/bean/mock/PortablePreferenceMockBeanGeneratedImpl.java]: Minor format issue (space missing after "(").

File: uberfire-preferences/uberfire-preferences-api/src/main/java/org/uberfire/preferences/shared/impl/DefaultPreferenceScopeResolutionStrategy.java [hunk 4]
```diff
@@ -41,7 +41,7 @@ protected DefaultPreferenceScopeResolutionStrategy() {
 
     public DefaultPreferenceScopeResolutionStrategy( final PreferenceScopeFactory scopeFactory,
                                                      final String componentKey ) {
-        final List<PreferenceScope> order = getScopeOrder( scopeFactory, componentKey );
+        final List<PreferenceScope> order = getScopeOrder(scopeFactory, componentKey );
```
Comment (paulovmr) [inline uberfire-preferences/uberfire-preferences-api/src/main/java/org/uberfire/preferences/shared/impl/DefaultPreferenceScopeResolutionStrategy.java]: Minor format issue (space missing after "(").

File: uberfire-preferences/uberfire-preferences-api/src/main/java/org/uberfire/preferences/shared/impl/PreferenceScopeFactoryImpl.java [hunk 5]
```diff
@@ -39,7 +39,7 @@ public PreferenceScopeFactoryImpl( @Customizable final PreferenceScopeTypes scop
     }
 
     @Override
-    public PreferenceScope createScope( final String type ) throws InvalidPreferenceScopeException {
+    public PreferenceScope createScope(final String type ) throws InvalidPreferenceScopeException {
```
Comment (paulovmr) [inline uberfire-preferences/uberfire-preferences-api/src/main/java/org/uberfire/preferences/shared/impl/PreferenceScopeFactoryImpl.java]: Minor format issue (space missing after "(").

File: uberfire-preferences/uberfire-preferences-api/src/test/java/org/uberfire/preferences/shared/impl/DefaultPreferenceScopeResolutionStrategyTest.java [hunk 6]
```diff
@@ -38,7 +37,7 @@
 
     @Before
     public void setup() {
-        final SessionInfoMock sessionInfo = new SessionInfoMock( userScopeKey );
+        final SessionInfoMock sessionInfo = new SessionInfoMock(DefaultPreferenceScopesForTests.userScopeKey );
```
Comment (paulovmr) [inline uberfire-preferences/uberfire-preferences-api/src/test/java/org/uberfire/preferences/shared/impl/DefaultPreferenceScopeResolutionStrategyTest.java]: Minor format issue (space missing after "("). Other occurences in this PR, I will stopped commenting all of them, please check your formatter. :)

File: uberfire-preferences/uberfire-preferences-client/src/main/resources/org/uberfire/preferences/client/resources/i18n/Constants.properties [hunk 7]
```diff
@@ -0,0 +1,25 @@
+#
```
Comment (paulovmr) [inline uberfire-preferences/uberfire-preferences-client/src/main/resources/org/uberfire/preferences/client/resources/i18n/Constants.properties]: I think all this translations could exist only at the `uberfire-preferences-ui-client`, since all of them appear to be referring to files in that project.

**Comment by manstis** [comment]
As an aside, re: formatting; QE are being particularly vocal about formatting resulting in a new "standard" style in droolsjbpm-build-bootstrap to use in IDEA. Personally I like the spaces but the " common opinion" was that Oracle's coding style does not have spaces and therefore neither should we. I tried to argue for spaces but gave up as others had stronger opinions.... so use the new "standard" and love the results (I get the impression QE want to introduce failing PRs if code formatting is violated in the future). Feel free to raise on BSIG.

**Comment by adrielparedes** [comment]
@paulovmr ready! So what are the next steps now? How can I help with the other PRs?

**Comment by csadilek** [comment]
jenkins restest this.
